src/coerced.rs

```diff
@@ -0,0 +1,30 @@
+pub trait Coerced<F> {
```

Comment:
I think this means the same as [`trait AsPrimitive`](https://docs.rs/num-traits/latest/num_traits/cast/trait.AsPrimitive.html)?